### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Taras2552**
- [DEMO LINK](https://Taras2552.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://Taras2552.github.io/layout_product-cards/report/html_report/)

- [ ] there 2 BEM blocks `card` and `stars` each in it own file
- [ ] SCSS Nesting is used for `elements`, `modifiers` and `pseudo-classes`
- [ ] SCSS Variables are used for main values and placed in a **separate file**
- [ ] all `stars--N` modifiers work as expected (Highlight first `N` stars)
- [ ] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -2,12 +2,45 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
     <link rel="stylesheet" href="./styles/index.scss">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <img
+        src="images/imac.jpeg"
+        alt="imac_images"
+        class="card__img">
```
Comment (anastasiia-tilikina) [inline src/index.html:17]:
```suggestion
      <img
        src="images/imac.jpeg"
        alt="imac_images"
        class="card__img"
      >
```

pay attention to code formatting

File: src/styles/index.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
-body {
+ * {
```
Comment (anastasiia-tilikina) [inline src/styles/index.scss:1]: don't use * selector for reseting margings. use it only for box-sizing property

File: src/styles/stars.scss [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  background-image: url("../images/star-active.svg");
+}
+
+.stars--4 :nth-child(-n+4) {
+  background-image: url("../images/star-active.svg");
+}
+
+.stars--5 :nth-child(-n+5) {
+  background-image: url("../images/star-active.svg");
+}
```
Comment (anastasiia-tilikina) [inline src/styles/stars.scss:30]:
```suggestion
.stars--1 :nth-child(-n+1) {
  background-image: url("../images/star-active.svg");
}

.stars--2 :nth-child(-n+2) {
  background-image: url("../images/star-active.svg");
}

.stars--3 :nth-child(-n+3) {
  background-image: url("../images/star-active.svg");
}

.stars--4 :nth-child(-n+4) {
  background-image: url("../images/star-active.svg");
}

.stars--5 :nth-child(-n+5) {
  background-image: url("../images/star-active.svg");
}
```

combine these selectors with coma

File: src/styles/stars.scss [hunk 4]
```diff
@@ -0,0 +1,30 @@
+.stars {
+  display: flex;
+}
+
+.stars__star {
+  background: url("../images/star.svg") center no-repeat;
+  height: 16px;
+  width: 16px;
+  margin-right: 4px;
```
Comment (anastasiia-tilikina) [inline src/styles/stars.scss:9]: reset margin for last child

File: src/styles/card.scss [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  padding: 32px 16px 17px;
+
+  &__img {
+    height: 134px;
+    width: 160px;
+    align-self: center;
+  }
+
+  &__title {
+    padding: 40px 0 4px;
```
Comment (anastasiia-tilikina) [inline src/styles/card.scss:18]: it is better to use margin for setting spasing between elements.
and be consistent use either margin top or margin bottom everywhere

File: src/styles/index.scss (hunk 2 above)
Comment (anastasiia-tilikina) [reply src/styles/index.scss:1]: not fixed here

File: src/index.html [hunk 6]
```diff
@@ -2,12 +2,45 @@
...
+
+      <div class="price">
+        <p class="price__title">Price:</p>
+        <p class="price__subtitle">$2,199</p>
+      </div>
+
+      <button class="card__button" data-qa="hover">
+         BUY
+      </button>
+
```
[the remark this answers is not in the document]
Comment (anastasiia-tilikina) [reply src/index.html:44]: not fixed

File: src/index.html [hunk 7]
```diff
@@ -2,12 +2,46 @@
...
     <link rel="stylesheet" href="./styles/index.scss">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <img
+        src="images/imac.jpeg"
+        alt="imac_images"
+        class="card__img"
+        >
```
Comment (anastasiia-tilikina) [inline src/index.html:18]:
```suggestion
      <img
        src="images/imac.jpeg"
        alt="imac_images"
        class="card__img"
      >
```
not exactly)
please double-check yourself with [style guide](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

File: src/styles/card.scss (hunk 5 above)
Comment (anastasiia-tilikina) [reply src/styles/card.scss:18]: not fixed

File: src/styles/stars.scss [hunk 8]
```diff
@@ -0,0 +1,18 @@
+.stars {
+  display: flex;
+  flex-direction: row;
```
Comment (anastasiia-tilikina) [inline src/styles/stars.scss:3]: this is default value

File: src/styles/stars.scss [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+.stars__star {
+  background: url("../images/star.svg") center no-repeat;
+  height: 16px;
+  width: 16px;
+
```
Comment (anastasiia-tilikina) [inline src/styles/stars.scss:10]: you need to reset margin for only last child, not remove it at all

File: src/index.html [hunk 10]
```diff
@@ -2,12 +2,46 @@
...
+      <h2 class="card__title">APPLE A1419 iMac 27" Retina <br> 5K Monoblock (MNED2UA/A)</h2>
+      <p class="card__id">Product code: 195434</p>
+
+      <div class="grade">
```
Comment (anastasiia-tilikina) [inline src/index.html:22]: this is not a separate BEM block but an element of card block. use proper class name

File: src/index.html [hunk 11]
```diff
@@ -2,12 +2,46 @@
...
+        <div class="stars stars--4">
```
Comment (anastasiia-tilikina) [inline src/index.html:23]: stars is also an element of card block use mix here

File: src/index.html [hunk 12]
```diff
@@ -2,12 +2,46 @@
...
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+        </div>
+
+        <div class="grade__reviews">
+          <p class="grade__text">Reviews: 5</p>
+        </div>
+      </div>
+
+      <div class="price">
```
Comment (anastasiia-tilikina) [inline src/index.html:36]: same about price

**Review by anastasiia-tilikina — CHANGES_REQUESTED** [review]
Please make sure that you've fixed all issues

File: src/styles/stars.scss [hunk 13]
```diff
@@ -0,0 +1,14 @@
+.stars {
+  display: flex;
+}
+
+.stars__star {
+  background: url("../images/star.svg") center no-repeat;
+}
```
Comment (etojeDenys) [inline src/styles/stars.scss:7]: use sass nesting here

File: src/styles/card.scss [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    color: $button-color;
+    border: 1px solid $button-color;
+  }
+}
+
+.card__grade {
+  display: flex;
+  justify-content: space-between;
+  padding: 16px 0 24px;
+}
```
Comment (etojeDenys) [inline src/styles/card.scss:56]: same here

File: src/index.html (hunk 10 above)
Comment (etojeDenys) [reply src/index.html:21]: image

File: src/index.html (hunk 12 above)
Comment (etojeDenys) [reply src/index.html:33]: image

File: src/index.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (etojeDenys) [reply src/index.html:44]: image

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/styles/stars.scss (hunk 13 above)
Comment (TarasHoliuk) [reply src/styles/stars.scss:7]: not fixed:
image

File: src/styles/stars.scss [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+
+.stars--1 :nth-child(-n+1),
+.stars--2 :nth-child(-n+2),
+.stars--3 :nth-child(-n+3),
+.stars--4 :nth-child(-n+4)  {
+background-image: url("../images/star-active.svg");
+}
```
Comment (TarasHoliuk) [inline src/styles/stars.scss:14]:
```suggestion
.stars--1 :nth-child(-n+1),
.stars--2 :nth-child(-n+2),
.stars--3 :nth-child(-n+3),
.stars--4 :nth-child(-n+4)  {
  background-image: url("../images/star-active.svg");
}
```
indentation - pay attention to code style

File: src/styles/card.scss [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+    color: $main-text-color;
+    font-size: 12px;
+    font-weight: 500;
+    line-height: 18px;
+  }
+
+  &__id {
+    color: $secondary-text-color;
+    font-size: 10px;
+    font-weight: 400;
```
Comment (TarasHoliuk) [inline src/styles/card.scss:28]: it's default value - no need to set it

File: src/index.html [hunk 17]
```diff
@@ -2,12 +2,42 @@
...
+      <div class="card__price">
+        <p class="card__price-text">Price:</p>
+        <p class="card__price-subtitle">$2,199</p>
+      </div>
+      <button
+            class="card__button"
+            data-qa="hover"
+      >
+           BUY
+      </button>
```
Comment (TarasHoliuk) [inline src/index.html:40]:
```suggestion
      <button class="card__button" data-qa="hover">
        BUY
      </button>
```
Read [Code Style Rules](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

File: src/styles/card.scss [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+    line-height: 14px;
+  }
+
+  &__reviews {
```
Comment (TarasHoliuk) [inline src/styles/card.scss:32]: Have you deployed the latest changes?

image

You always should run `npm run deploy` after any changes. [Instruction](https://mate-academy.github.io/layout_task-guideline/#codenpm-run-deploycode-to-publish-to-github-pages), point 20

File: src/styles/card.scss [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+    display: flex;
+    justify-content: space-between;
+    padding: 16px 0 24px;
```
Comment (TarasHoliuk) [inline src/styles/card.scss:35]: it's better to set only the top or only the bottom margin/padding.

in this case, it's more `margin` than `padding` - it's outer indentation

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
Check how the deployed version looks:

image

Make sure it matches the design (the same about tests).

**Pull request closed without merging**